src/usePermission.ts

```diff
@@ -30,13 +30,16 @@ const usePermission = (permissionDesc: PermissionDesc): State => {
   };
 
   useEffect(() => {
-    navigator.permissions
-      .query(permissionDesc)
-      .then(status => {
-        permissionStatus = status;
-        changeState();
-      })
-      .catch(noop);
+    // Permissions API is not available in every browser
+    if (navigator.permissions && navigator.permissions.query) {
+      navigator.permissions
+        .query(permissionDesc)
+        .then((status) => {
+          permissionStatus = status;
+          changeState();
+        })
+        .catch(noop);
+    }
```

Comment:
If the API is not supported, then we will not know about it. How about making a special status `unavailable` here?
```
} else {
    permissionStatus = 'unavailable';
    changeState();
}
```

Reply:
I tried my example and understand that this is not the best solution. 
Better this way
```
} else {
    setState('unavailable');
}
```